Added ability to request a specfic page via URL

dlashua opened the pull request:
A URL like:

http://127.0.0.1:8123/local/TileBoard/index.html?blargh

will cause TileBoard to open on a page with an "id" of "blargh".

This is useful if you have mounted panels in multiple rooms and would like each room to start TileBoard on a page devoted to that room.

rchl commented:
* You should use key=value pair for that functionality (for example pageid=foo). People might be currently using some url params already for some functionality. At least I'm using one already for specifying mobile browser.

* the `page.id` key is something you have introduced right? Should add documentation for it.

* If you are calling the key `id` then why comparing it with variable named `desiredPageName`? Doesn't seem to make sense.

* You should update URL state on changing pages. I think it makes sense to remember last set page rather than have it apply for ever after page was loaded.

rchl commented on scripts/controllers/main.js:
```diff
@@ -3,6 +3,8 @@ App.controller('Main', ['$scope', MainController]);
    if(!window.CONFIG) return;
 
+
+
```
Avoid random changes like that. :)

dlashua commented:
Modified to use pageid= query string parameter.

Parsed QueryString the "hard way" in order to avoid relatively new URLSearchParams class, just in case this is not desirable. 

Modified internal variable name from desiredPageName to desiredPageID

page.id was not introduced by me. It is referenced in the event section in regard to setting up a home assistant event to call $scope.openPage().

Updating the URL to record the last page being viewed could be useful if one presses "reload" or attempts to bookmark a page or share a URL. However, my limited knowledge of javascript in browsers seems to remember that the only way to do this without forcing an actual page reload (which is not needed) would be to use an anchor (#pageid) which TileBoard doesn't seem to like for some reason (?).

rchl commented:
> page.id was not introduced by me. It is referenced in the event section in regard to setting up a home assistant event to call $scope.openPage().

OK. I see it mentioned in the documentation but as far as I can see, TileBoard doesn't set the id, one has to add it manually to his/her config. It would be good to document this new functionality briefly I think.

> Updating the URL to record the last page being viewed could be useful if one presses "reload" or attempts to bookmark a page or share a URL.

We should think about it and probably address it right away. My use case is a bit different -- on mobile, the browser will evict the page from memory when browser is in the background. That means that TileBoard will reload and revert to initial page. I would like it to stay on last opened page instead.
With your fix it would always revert to initially defined page. It might or might not be expected depending on the use case.
(This problem is even more pronounced when using TileBoard as installed web app. Then it's removed from memory almost immediately after putting it in the background.)

> However, my limited knowledge of javascript in browsers seems to remember that the only way to do this without forcing an actual page reload (which is not needed) would be to use an anchor (#pageid) which TileBoard doesn't seem to like for some reason (?).

It seems that angularjs has some built-in minimal handling that makes that happen. Proper solution could be adding a router (either using angular's ngRoute or external uiRouter), but maybe that's too much.

resoai commented:
This is just making a mess and is not really relevant. All we need is to add "default" attribute to the page and You can  decide how to mark one as default in the config.
